scheduler/src/cook/rest/api.clj

```diff
@@ -866,7 +879,13 @@
                     executor (assoc :job/executor executor)
                     progress-output-file (assoc :job/progress-output-file progress-output-file)
                     progress-regex-string (assoc :job/progress-regex-string progress-regex-string)
-                    pool (assoc :job/pool (:db/id pool))
+                    pool-name-from-submission
```

Comment:
You're checking pool-name-from-submission as well as associating job-pool here. That mismatch seems weird and worth explaining why you're using submisison pool to gate this, but inserting a different pool. Please put in a comment.

Reply:
Added a comment, good call